src/daft-sql/src/planner.rs

```diff
@@ -1458,11 +1458,11 @@ impl<'a> SQLPlanner<'a> {
                 let start = self.plan_expr(substring_from)?;
                 let length = self.plan_expr(substring_for)?;
 
+                // SQL substring is one indexed
+                let start = start.sub(lit(1));
+
                 Ok(daft_functions::utf8::substr(expr, start, length))
             }
-            SQLExpr::Substring { special: false, .. } => {
-                unsupported_sql_err!("`SUBSTRING(expr [FROM start] [FOR len])` syntax")
-            }
```

Comment:
i don't know if we want/need to support both syntaxes. I think the `SUBSTRING(expr, start, length)` is definitely the more common one. Why do we need to add support for the other?

Reply:
Ah, I just added it because I noticed that they should have the same behavior anyway. But if we do not want both syntaxes I will remove this one.